tests/testsuite/mock-std/library/proc_macro/src/lib.rs

```diff
@@ -1,10 +1,15 @@
 #![feature(staged_api)]
 #![stable(since = "1.0.0", feature = "dummy")]
 
+
 extern crate proc_macro;
 
+// Don't re-export everything in the root so that the mock std can be distinguished from the real one.
 #[stable(since = "1.0.0", feature = "dummy")]
-pub use proc_macro::*;
+pub mod exported {
+    #[stable(since = "1.0.0", feature = "dummy")]
+    pub use proc_macro::*;
+}
```

Comment:
Can you clarify why this is needed?

In general, I'm not seeing the connection with proc-macro changes and support for per-pkg-target.